plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -171,6 +171,7 @@ public final class IcebergUtil
     public static final String METADATA_FILE_EXTENSION = ".metadata.json";
     private static final Pattern SIMPLE_NAME = Pattern.compile("[a-z][a-z0-9]*");
     static final String TRINO_QUERY_ID_NAME = "trino_query_id";
+    private static final Pattern HADDOP_GENERATED_VERSION_PATTERN = Pattern.compile("v([^\\.]*)\\..*");
```

Comment:
Hadoop catalog puts `version-hint.text` file under the metadata directory as you may already know. Allowing the registering & writing to such tables means the version becomes obsolete. Do you know how other query engines use the file? 

Also, the table may contain `iceberg.catalog` table property. Should we remove the property when registering the table?

Reply:
> Do you know how other query engines use the file?

Spark

https://github.com/apache/iceberg/blob/f5346cacf17d63257ab433215e1661ed221750d0/core/src/main/java/org/apache/iceberg/ReachableFileUtil.java#L48-L53

Reply:
I already saw the code before leaving the above comment. I'm asking other query engines' **_behavior_**, not the method place.

Reply:
> Also, the table may contain iceberg.catalog table property. Should we exclude the property when registering the table?

What negative implications do you see on not removing the property?

I'm thinking if a table is to be registered for _read_ operations in Trino, but _write_ still continues via Spark on Hadoop, we would break the table on Hadoop's side.

Reply:
When creating an Iceberg table with the Hadoop Catalog/HadoopTable API, there is no need for a metastore like in the case of HiveCatalog/GlueCatalog, which require a metastore. In catalog-based tables, the `metadata_location` property is used (found in the metastore) to locate the latest metadata file and extract its version to write the next metadata file and then update `metadata_location` property atomically. On the other hand, Hadoop Catalog/Hadoop Table uses the `version-hint.txt` file to determine the most recent metadata version. It utilizes this information to create the next metadata file and then updates the `version-hint.txt` file with the new version in an atomic manner.

>  writing to such tables means the version becomes obsolete

Yes, When registering the table in Trino, `version-hint.txt` becomes obsolete